### ISSUE TITLE
RangeError (index): Invalid value: Not in inclusive range 0..4: 6

### ISSUE THREAD

**Issue by felixjunghans** [Magnuti/Weekly-Date-Picker#6]
Hey, thanks for the great package. However, I have an error in my app that currently occurs. 

This is the exception: 

RangeError (index): Invalid value: Not in inclusive range 0..4: 6

When the exception was thrown, this was the stack: 
#0      List.[] (dart:core-patch/growable_array.dart:264:36)
#1      _WeeklyDatePickerState._dateButton (package:weekly_date_picker/weekly_date_picker.dart:161:43)
#2      _WeeklyDatePickerState._weekdays (package:weekly_date_picker/weekly_date_picker.dart:152:20)

I did some debugging and found the cause. On 29.10.2023 is time change in Europe. 

The following code is the problem
```
     final int offset = i + 1 - _initialSelectedDay.weekday;
      final int daysToAdd = weekIndex * 7 + offset;
      final DateTime dateTime =
          _initialSelectedDay.add(Duration(days: daysToAdd));
      weekdays.add(_dateButton(dateTime));
```

At  _initialSelectedDay.add(Duration(days: daysToAdd)); when  2023-10-10 00:00:00.000 is _initialSelectedDay and daysToAdd is 20 Dart does not return 2023-10-30 00:00:00.000 as expected but 2023-10-29 23:00:00.000. 

I think this is a time zone or time conversion problem.

I tried this code in DartPad to validate the Exception and it also returns 2023-10-29 instead of 2023-10-30 

```
void main() {
     var initialDate = new DateTime(2023, 10, 10);
     final newDate = initialDate.add(Duration(days: 20));
     print(newDate);
}
```

**Comment by Magnuti** [comment]
Thank you for the issue @felixjunghans! Unfortunately this seems out of my hands. It seems like a bug in the Dart language. I crossed checked this with Python (and my calendar lol) and 20 days from 10.10.2023 is 30.10.2023.

<img width="879" alt="image" src="https://github.com/Magnuti/Weekly-Date-Picker/assets/43852515/7ce43c61-490e-4024-b9e9-37e7492c3ce6">

Could you please submit a new issue to the Dart language's GitHub?

**Comment by Magnuti** [comment]
Nevertheless, thank you for the issue again @felixjunghans, I will look into a quick fix in the meantime in case the Flutter/Dart team isn't quick enough. Please add a link to the new issue here if you create one.

**Comment by felixjunghans** [comment]
Hey @Magnuti thank you for your quick response. I was about to create a new issue in dart-lang but then I found out that it looks like it is the desired behavior. At least that's how it's described in this issue [https://github.com/dart-lang/sdk/issues/37449](https://github.com/dart-lang/sdk/issues/37449). 

So it would be cool if you can find a workaround to fix this in the package :) Thanks a lot.

**Comment by Magnuti** [comment]
Thank you! I see now that the issue is also described properly in the [documentation](https://api.flutter.dev/flutter/dart-core/DateTime/add.html) of `DateTime.add()` as you mentioned.

<img width="615" alt="image" src="https://github.com/Magnuti/Weekly-Date-Picker/assets/43852515/e245c58d-79d0-411e-9da3-41a2db43df7b">

I am working on a fix as we speak.

**Comment by Magnuti** [comment]
This has now been patched in version 1.3.2 https://pub.dev/packages/weekly_date_picker/versions/1.3.2. Thank you again for making me aware of this @felixjunghans!

**Comment by felixjunghans** [comment]
Wow thanks for fixing it so quickly.
